src/main/java/com/palantir/gradle/gitversion/GitVersionPlugin.java

```diff
@@ -16,38 +16,56 @@
 
 package com.palantir.gradle.gitversion;
 
+import com.google.common.collect.ImmutableMap;
 import groovy.lang.Closure;
-import java.io.File;
-import java.io.IOException;
-import org.eclipse.jgit.api.Git;
-import org.eclipse.jgit.internal.storage.file.FileRepository;
 import org.gradle.api.Action;
 import org.gradle.api.Plugin;
 import org.gradle.api.Project;
 import org.gradle.api.Task;
+import org.gradle.api.provider.Provider;
 
 public final class GitVersionPlugin implements Plugin<Project> {
-    private final Timer timer = new Timer();
 
     @Override
     public void apply(final Project project) {
-        project.getRootProject().getPluginManager().apply(GitVersionRootPlugin.class);
+        int maxParallelUsage = 2;
+        Provider<GitVersionCacheService> serviceProvider = project.getGradle()
+                .getSharedServices()
+                .registerIfAbsent("GitVersionCacheService", GitVersionCacheService.class, spec -> {
+                    // Provide some parameters
+                    spec.getMaxParallelUsages().set(maxParallelUsage);
```

Comment:
Do we need to set a max parallel usages at all? I think provided we make the class thread safe, it should be ok to allow max parallel access.

Reply:
I removed this line.